### PR TITLE
TCL: Remove TCL dependency from Alarms

### PR DESCRIPTION
**Pull request by chitianhao**

### CODE REVIEW

**Comment by zwoop** [comment]
I think we should aim to collect all these TCL removals for 8.1.0

**Comment by chitianhao** [comment]
Yes, we should. I'll also make sure they don't break anything.

File: mgmt/Alarms.cc [hunk 1]
```diff
@@ -135,15 +133,17 @@ Alarms::resolveAlarm(alarm_t a, char *ip)
...
-    ink_hash_table_delete(local_alarms, buf);
-    ats_free(((Alarm *)hash_value)->description);
+  if (!ip && local_alarms.find(buf) != local_alarms.end()) {
+    Alarm *hash_value = local_alarms[buf];
+    local_alarms.erase(buf);
+    ats_free(hash_value->description);
     ats_free(hash_value);
-  } else if (ip && ink_hash_table_lookup(remote_alarms, buf, &hash_value) != 0) {
+  } else if (ip && remote_alarms.find(buf) != remote_alarms.end()) {
     char buf2[1024];
```
Comment (SolidWallOfCode) [inline mgmt/Alarms.cc]: I realize this is outside the scope of the PR, but what the heck is the point of `buf2`?

File: mgmt/Alarms.cc [hunk 2]
```diff
@@ -310,16 +306,11 @@ Alarms::signalAlarm(alarm_t a, const char *desc, const char *ip)
...
-
   ink_mutex_acquire(&mutex);
-  for (entry = ink_hash_table_iterator_first(remote_alarms, &iterator_state); entry != nullptr;
-       entry = ink_hash_table_iterator_next(remote_alarms, &iterator_state)) {
-    char *key  = (char *)ink_hash_table_entry_key(remote_alarms, entry);
-    Alarm *tmp = (Alarm *)ink_hash_table_entry_value(remote_alarms, entry);
-
-    if (strstr(key, ip)) {
+  for (auto &&it : remote_alarms) {
+    std::string key = it.first;
```
Comment (SolidWallOfCode) [inline mgmt/Alarms.cc]: `std::string const&`.

File: mgmt/Alarms.cc [hunk 3]
```diff
@@ -335,18 +326,14 @@ Alarms::resetSeenFlag(char *ip)
...
-  for (entry = ink_hash_table_iterator_first(remote_alarms, &iterator_state); entry != nullptr;
-       entry = ink_hash_table_iterator_next(remote_alarms, &iterator_state)) {
-    char *key  = (char *)ink_hash_table_entry_key(remote_alarms, entry);
-    Alarm *tmp = (Alarm *)ink_hash_table_entry_value(remote_alarms, entry);
-
-    if (strstr(key, ip)) {                         /* Make sure alarm is for correct ip */
-      if (!tmp->seen) {                            /* Make sure we did not see it in peer's report */
-        ink_hash_table_delete(remote_alarms, key); /* Safe in iterator? */
+  for (auto &&it : remote_alarms) {
+    std::string key = it.first;
```
Comment (SolidWallOfCode) [inline mgmt/Alarms.cc]: `std::string const& key`

File: mgmt/api/CoreAPI.cc [hunk 4]
```diff
@@ -803,15 +799,12 @@ ActiveEventGetMlt(LLQ *active_events)
...
   // value = alarm_description defined in Alarms.cc alarmText[] array
-  event_ht = lmgmt->alarm_keeper->getLocalAlarms();
+  std::unordered_map<std::string, Alarm *> const &event_ht = lmgmt->alarm_keeper->getLocalAlarms();
 
   // iterate through hash-table and insert event_name's into active_events list
-  for (entry = ink_hash_table_iterator_first(event_ht, &iterator_state); entry != nullptr;
-       entry = ink_hash_table_iterator_next(event_ht, &iterator_state)) {
-    char *key = (char *)ink_hash_table_entry_key(event_ht, entry);
-
+  for (auto it : event_ht) {
```
Comment (SolidWallOfCode) [inline mgmt/api/CoreAPI.cc]: `auto &&`

File: mgmt/Alarms.cc (hunk 1 above)
Comment (chitianhao) [reply mgmt/Alarms.cc]: Right. It makes no sense. Maybe someone forgot to remove it when they made changes here.

**Comment by bryancall** [comment]
Cherry picked to 8.1.x

**Pull request merged**